src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Reference.cs

```diff
@@ -89,40 +89,24 @@ public void Release(object obj)
     /// <summary>
     /// Used to cache a weak reference in a threadsafe way
     /// </summary>
-    internal sealed class SharedReference
+    internal sealed class SharedReference<T> where T : class
```

Comment:
With the locking removed, this class is useless. It can be just replaced by `WeakReference<T>`.

Reply:
Yes, I already removed it but didn't push it here as I'll probably close this PR because I incorporated this in another one. Thanks